### PR TITLE
Dashboard

### LINKED ISSUE #5: Dashboard
As a Student
I want to have my personal dashboard
where I know I'll find anything I need.

## Acceptance Criteria
- [ ] On a new Page, the logged in Users name is displayed with a greeting

## Tasks
- [ ] Create tests
- [ ] Create the Dashboard Page
- [ ] Add Styling

### PR DESCRIPTION
**Pull request by pixxelschubser**
To resolve #5 I added React Router to the project. The user can now log into the App and is then redirected to a dashboard, where the users name is displayed.
### Add
- React Router
- Front Page
- Dashboard Page
- Greeting Component
### Edit
- Login Form
- Logo Component
- Button Component
### Issues Resolved
- #5

### CODE REVIEW

File: src/components/Greeting.spec.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { render, screen } from "@testing-library/react";
+import Greeting from "./Greeting";
+
+describe('Greetig', () => {
```
Comment (mortaen) [inline src/components/Greeting.spec.js:4]: "Greetig"  is missing a letter 😄

File: src/App.js [hunk 2]
```diff
@@ -1,27 +1,44 @@
-import styled from "styled-components/macro";
-import logo from './assets/images/student-compass_logo.svg';
-import LoginForm from './components/LoginForm.js';
+import { 
+  BrowserRouter as Router,
```
Comment (macfrei) [inline src/App.js:2]: I like it better to have it in the `index.js` but that is just my personal preference :)

File: src/App.js [hunk 3]
```diff
@@ -1,27 +1,44 @@
...
+          {authenticated 
+            ? <Redirect to="/dashboard" /> 
+            : <Frontpage onSubmit={onLogin}/>
+          }
+        </Route>
+        <Route path="/dashboard">
+          {authenticated
+            ? <Dashboard username={userName}/>
+            : <Redirect to="/" />
+          }
```
[the remark this answers is not in the document]
Comment (macfrei) [reply src/App.js:39]: He's redirecting to the path `"/"` which either redirects to `"/dashboard"` or renders the `FrontPage`. So he basically is doing what you are asking 🙂 the thing is that the `FrontPage` is probably the log in and this should be on the path `"/"`. If he'd do it as you suggests, both `Dashboard` and `FrontPage` would be rendered at the path `"/dashboard"`.

File: src/components/Greeting.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import styled from "styled-components/macro";
+
+export default function Greeting({greeting, username}){
+
+	let name = username[0].toUpperCase() + username.slice(1)
+	name = name.substring(0, name.indexOf("."))
```
Comment (macfrei) [inline src/components/Greeting.js:6]: Maybe add that to a separate function? 
```suggestion
function getName(username) {
	const name = username[0].toUpperCase() + username.slice(1)
	return name.substring(0, name.indexOf("."))
}

const name = getName(username)
```

Don't know why, but for me that looks a bit neater. But as usual, that is just my opinion!
(you could even extract that function and write a little unit test for it! 😃

File: src/components/Greeting.spec.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	it('renders a greeting followed by a given name', () => {
+		const username = "monika.musterfrau@teacher"
+		const greeting = "Hallo"
+		render(<Greeting greeting={greeting} username={username} />)
+
+		const message = screen.getByText("Hallo Monika.")
+		expect(message).toBeInTheDocument();
```
Comment (macfrei) [inline src/components/Greeting.spec.js:11]: Nice!

File: src/components/Logo.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import styled from "styled-components/macro";
+
+export default function Logo() {
+	return(
+		<Image 
+			src="/assets/images/student-compass_logo.svg" alt="Student Compass"
```
Comment (macfrei) [inline src/components/Logo.js:6]: Nice, good use of an `alt` text!

**Pull request merged**